ui/src/app/components/gke-app-configuration-panels/create-gke-app.spec.tsx

```diff
@@ -288,6 +299,218 @@ describe(CreateGkeApp.name, () => {
         // Assert that the autodeleteRemainingDays text is not in the document
         expect(autodeleteRemainingDaysText).not.toBeInTheDocument();
       });
+
+      // positive tests for machine type configuration
+
+      it.each([
+        ['there are no running apps', {}],
+        [
+          'the current app is DELETED',
+          {
+            userApps: [{ ...listAppsResponse(), status: AppStatus.DELETED }],
+          },
+        ],
+      ])(
+        `should allow machine type configuration when %s`,
+        async (_, propOverrides: Partial<CreateGkeAppProps>) => {
+          serverConfigStore.set({
+            config: {
+              ...serverConfigStore.get().config,
+              enableGKEAppMachineTypeChoice: true,
+            },
+          });
+
+          const { container } = await component(appType, propOverrides);
+
+          // default is 4 CPUs, 15 GB RAM, n1-standard-4
```

Comment:
Should we add check for defaults?

Reply:
good idea.  I added the "should use the default machine type" test, and also added a check to each of the "should not allow machine type configuration" cases